File: app/src/main/java/ru/neosvet/vestnewage/service/CheckStarter.java
package ru.neosvet.vestnewage.service;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.work.Constraints;
import androidx.work.NetworkType;
import androidx.work.PeriodicWorkRequest;
import androidx.work.WorkManager;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import java.util.concurrent.TimeUnit;

import ru.neosvet.utils.Const;
import ru.neosvet.vestnewage.App;

public class CheckStarter extends Worker {
    private static final String TAG_PERIODIC = "check periodic";

    public static void set(int time) {
        WorkManager work = WorkManager.getInstance(App.context);
        work.cancelAllWorkByTag(TAG_PERIODIC);
        if (time == Const.TURN_OFF) return;
        Constraints constraints = new Constraints.Builder()
                .setRequiredNetworkType(NetworkType.CONNECTED)
                .setRequiresBatteryNotLow(true)
                .build();
        PeriodicWorkRequest task = new PeriodicWorkRequest
                .Builder(CheckStarter.class, time, TimeUnit.MINUTES, time - 5, TimeUnit.MINUTES)
                .setConstraints(constraints)
                .addTag(TAG_PERIODIC)
                .build();
        work.enqueue(task);
    }

    public CheckStarter(@NonNull Context context, @NonNull WorkerParameters workerParams) {
        super(context, workerParams);
    }

    @NonNull
    @Override
    public Result doWork() {
        CheckService.postCommand(true);
        return Result.success();
    }
}

File: app/src/main/java/ru/neosvet/vestnewage/service/CheckStarter.kt
package ru.neosvet.vestnewage.service

import android.content.Context
import androidx.work.*
import ru.neosvet.vestnewage.App
import ru.neosvet.vestnewage.utils.Const
import java.util.concurrent.TimeUnit

class CheckStarter(
    context: Context, workerParams: WorkerParameters
) : Worker(context, workerParams) {

    override fun doWork(): Result {
        CheckService.postCommand(true)
        return Result.success()
    }

    companion object {
        private const val TAG_PERIODIC = "check periodic"
        @JvmStatic
        fun set(time: Int) {
            val work = WorkManager.getInstance(App.context)
            work.cancelAllWorkByTag(TAG_PERIODIC)
            if (time == Const.TURN_OFF) return
            val constraints = Constraints.Builder()
                .setRequiredNetworkType(NetworkType.CONNECTED)
                .setRequiresBatteryNotLow(true)
                .build()
            val task = PeriodicWorkRequest.Builder(
                CheckStarter::class.java,
                time.toLong(),
                TimeUnit.MINUTES,
                (time - 5).toLong(),
                TimeUnit.MINUTES
            )
                .setConstraints(constraints)
                .addTag(TAG_PERIODIC)
                .build()
            work.enqueue(task)
        }
    }
}